lib/tmdb_handler.rb

```diff
@@ -81,52 +88,74 @@ def tmdb_handler_add_movie(tmdb_id)
     popularity: @movie.popularity, runtime: @movie.runtime, mpaa_rating: @movie.mpaa_rating)
   end
 
-  def tmdb_handler_update_movie(tmdb_id)
-    @movie_url = "https://api.themoviedb.org/3/movie/#{tmdb_id}?api_key=#{ENV['tmdb_api_key']}&append_to_response=trailers,credits,similar,releases"
-    @result = JSON.parse(open(@movie_url).read, symbolize_names: true)
-    @movie_to_update = Movie.find_by(tmdb_id: tmdb_id)
-    @movie = MovieMore.tmdb_info(@result)
-
-    @movie_to_update.update_attributes(title: @movie.title, imdb_id: @movie.imdb_id, genres: @movie.genres,
-      actors: @movie.actors, adult: @result[:adult], backdrop_path: @movie.backdrop_path,
-      poster_path: @movie.poster_path, release_date: @movie.release_date, overview: @movie.overview,
-      trailer: @movie.trailer, director: @movie.director, director_id: @movie.director_id,
-      vote_average: @movie.vote_average, popularity: @movie.popularity, runtime: @movie.runtime,
-      mpaa_rating: @movie.mpaa_rating)
+  def self.tmdb_handler_update_movie(tmdb_id)
+    movie = Movie.find_by(tmdb_id: tmdb_id)
+    raise TmdbHandlerError.new("Movie not found in DB: tmdb_id: #{tmdb_id}") unless movie
+
+    movie_url = "#{BASE_URL}/movie/#{tmdb_id}?api_key=#{ENV['tmdb_api_key']}&append_to_response=trailers,credits,similar,releases"
+    api_result = HTTParty.get(movie_url).deep_symbolize_keys rescue nil
+    raise TmdbHandlerError.new("API request failed for tmdb_id: #{tmdb_id}") unless api_result && api_result[:id].to_s == tmdb_id.to_s
+
+    updated_data = MovieMore.tmdb_info(api_result)
+
+    if movie.title != updated_data.title
+      msg = "Movie title doesn't match. tmdb_id: #{tmdb_id}. Current title: #{movie.title}. Updated title: #{updated_data.title}"
+      raise TmdbHandlerError.new(msg)
+    end
```

Comment:
Instead of raising an error, let's just return out of this method. This will skip the update for this particular movie, but allow the rake task to continue to run for other movies:

```ruby
if movie.title != updated_data.title
  puts "Movie title doesn't match. tmdb_id: #{tmdb_id}. Current title: #{movie.title}. Updated title: #{updated_data.title}"

  return
end
```